go/libraries/doltcore/sqle/dsess/dolt_session.go

```diff
@@ -128,12 +129,29 @@ func getPersistedValue(conf config.ReadableConfig, k string) (interface{}, error
 
 	var res interface{}
 	switch value.(type) {
-	case int, int8, int16, int32, int64:
+	case int8:
+		switch strings.ToLower(v) {
+		case "on", "true", "1":
```

Comment:
This is a little alarming. The engine layer should translate these values to 0 or 1 before they ever get to an integrator. There are steps in the analyzer where we do exactly that in other contexts.

Reply:
Is this to support someone doing `dolt config --add key true`?

Reply:
yeah, should we not support that? i can just funnel everything through int8